mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -64,99 +64,105 @@ class Completions(
+      case (_: Import) :: _ => CursorPos.Import
+      case _ :: (_: Import) :: _ => CursorPos.Import
       case (head: (Select | Ident)) :: tail =>
         // https://github.com/lampepfl/dotty/issues/15750
         // due to this issue in dotty, because of which trees after typer lose information,
         // we have to calculate hasNoSquareBracket manually:
-        val hasNoSquareBracket =
+        val hasSquareBracket =
           val span: Span = head.srcPos.span
           if span.exists then
             var i = span.end
             while i < (text.length() - 1) && text(i).isWhitespace do i = i + 1
 
-            if (i < text.length()) then text(i) != '['
-            else true
+            if (i < text.length()) then text(i) == '['
+            else false
           else false
+
+        def typePos = CursorPos.Type(hasSquareBracket, hasNewKw = false)
+        def newTypePos =
+          CursorPos.Type(hasSquareBracket, hasNewKw = true)
+
         tail match
-          case (v: ValOrDefDef) :: _ =>
-            if v.tpt.sourcePos.contains(pos) then
-              CursorPositionCondition(true, false, false, hasNoSquareBracket)
-            else
-              CursorPositionCondition(false, false, false, hasNoSquareBracket)
-          case New(selectOrIdent: (Select | Ident)) :: _ =>
-            if selectOrIdent.sourcePos.contains(pos) then
-              CursorPositionCondition(false, true, false, hasNoSquareBracket)
-            else
-              CursorPositionCondition(false, false, false, hasNoSquareBracket)
-          case (a @ AppliedTypeTree(_, args)) :: _ =>
-            if args.exists(_.sourcePos.contains(pos)) then
-              CursorPositionCondition(true, false, false, hasNoSquareBracket)
-            else CursorPositionCondition(false, false, false, false)
-          case (_: Import) :: _ =>
-            CursorPositionCondition(false, false, false, hasNoSquareBracket)
+          case (v: ValOrDefDef) :: _ if v.tpt.sourcePos.contains(pos) =>
+            typePos
+          case New(selectOrIdent: (Select | Ident)) :: _
+              if selectOrIdent.sourcePos.contains(pos) =>
+            newTypePos
+          case (a @ AppliedTypeTree(_, args)) :: _
+              if args.exists(_.sourcePos.contains(pos)) =>
+            typePos
+          case (Template(constr, parents, self, _)) :: _
+              if (constr :: self :: parents).exists(
+                _.sourcePos.contains(pos)
+              ) =>
+            typePos
           case _ =>
-            CursorPositionCondition(false, false, true, hasNoSquareBracket)
+            CursorPos.Term
         end match
 
       case (_: TypeTree) :: TypeApply(Select(newQualifier: New, _), _) :: _
           if newQualifier.sourcePos.contains(pos) =>
-        CursorPositionCondition(false, true, false, true)
+        CursorPos.Type(hasTypeParams = false, hasNewKw = true)
 
-      case _ => CursorPositionCondition(false, false, false, true)
+      case _ => CursorPos.Term
```

Comment:
why is this TermPos? It does not differentiate between `CursorPositionCondition(false, false, true, hasNoSquareBracket)` and this one